Code/Source/Lidar/LidarTemplateUtils.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "LidarTemplateUtils.h"
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/Math/MathUtils.h>
+#include <AzCore/Math/Matrix4x4.h>
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Utils/Utils.h>
+
+namespace ROS2
+{
+    LidarTemplate LidarTemplateUtils::GetTemplate(LidarTemplate::LidarModel model)
+    {
+        static std::unordered_map<LidarTemplate::LidarModel, LidarTemplate> templates;
+
+        if (templates.empty())
+        {
+            LidarTemplate generic3DLidar =
+            {
+                .m_model = LidarTemplate::Generic3DLidar,
```

Comment:
[Designated Initializers](https://en.cppreference.com/w/cpp/language/aggregate_initialization#Designated_initializers) isn't supported until C++20, so this code shouldn't be compiling.

Reply:
I am using clang 12, it does support designated initializers (since clang 10).
https://en.cppreference.com/w/cpp/compiler_support.
However, I understand that this is not universal among supported platforms and compilers.
Default member initialization is inferior as it is relatively easy to confuse the order or miss an error when extending struct. But if it is all we have - I will switch to it.

Reply:
Default Member initialization is initializing the class members as part of the struct definition.
That is different from the member initializer list which is used as part of the constructor.

By [default member initialization](https://en.cppreference.com/w/cpp/language/data_members#Member_initialization) it would look like the following
```
struct LidarTemplate
{
       LidarModel m_model;
        AZStd::string m_name;
        float m_minHAngle = 0.0f;
        float m_maxHAngle = 0.0f;
        float m_minVAngle = 0.0f;
        float m_maxVAngle = 0.0f;
        int m_layers = 0;
        int m_numberOfIncrements = 0;
        float m_maxRange = 0.0f;
}
```

Reply:
Oh I see, I thought you meant something else. This won't do as it does not help if there is more than one type of lidar.